### PR TITLE
Add mypy checker

### PR DESCRIPTION
**Pull request by leehuwuj**
## Summary by CodeRabbit

- **New Features**
	- Added a pre-commit hook for formatting checks on template files.
	- Enhanced dependency management for various model configurations.
	- Improved error handling and logging during initialization processes.

- **Bug Fixes**
	- Updated handling of environment variables to prevent potential errors.
	- Added checks to prevent errors in event handling logic.

- **Documentation**
	- Enhanced type annotations and comments for better clarity across multiple files.

- **Chores**
	- Introduced `mypy` for type-checking capabilities in the project.

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -1,2 +1,3 @@
 pnpm format
 pnpm lint
+uvx ruff format --check templates/
```
Comment (leehuwuj) [inline .husky/pre-commit:3]: Check this document to install uv and uvx cli: https://docs.astral.sh/uv/getting-started/installation/

File: templates/components/vectordbs/python/milvus/vectordb.py [hunk 2]
```diff
@@ -15,6 +16,6 @@ def get_vector_store():
         user=os.getenv("MILVUS_USERNAME"),
         password=os.getenv("MILVUS_PASSWORD"),
         collection_name=collection,
-        dim=int(os.getenv("EMBEDDING_DIM")),
+        dim=int(os.getenv("EMBEDDING_DIM", 768)),
```
Comment (marcusschiesser) [inline templates/components/vectordbs/python/milvus/vectordb.py:19]: Why no default?

Comment (leehuwuj) [reply templates/components/vectordbs/python/milvus/vectordb.py:19]: It's adding default value

File: templates/components/engines/python/chat/__init__.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+from .engine import get_chat_engine
```
[the remark this answers is not in the document]
Comment (leehuwuj) [reply templates/components/engines/python/chat/__init__.py]: thanks, this file is not needed now

**Pull request merged**